fastai/callbacks/hooks.py

```diff
@@ -181,6 +184,18 @@ def model_summary(m:Learner, n:int=70):
     res += f"\nTotal params: {total_params:,}\n"
     res += f"Total trainable params: {total_trainable_params:,}\n"
     res += f"Total non-trainable params: {total_params - total_trainable_params:,}\n"
+           
+    res += f"Optimized with {str(m.opt_func)[25:-1].replace('>', '')}\n"
+    if m.true_wd and "adam" in str(m.opt_func).lower():
```

Comment:
Don't test for Adam here, the behavior is different for all optimizers, even SGD when there is momentum (as discussed in the same article).

Reply:
Ok ! (I didn't understood that when reading the doc :\)